src/plugins/configuration.js

```diff
@@ -0,0 +1,99 @@
+/* @flow */
+
+import get from 'lodash/get';
+import merge from 'lodash/merge';
+import set from 'lodash/set';
+
+import Logger from '../logger';
+
+import type { PocoPluginType } from '../types';
+
+
+class ConfigurationService {
+    _environment: Object;
+
+    appName: string;
+
+    config: Object;
+
+    logger: Object;
+
+    constructor ({ appName, environment }) {
+        this.logger = new Logger('ConfigurationService');
+        this.logger.debug('constructor');
+
+        this.appName = appName;
+        this.config = get(environment, 'config') || {};
+        this._environment = environment;
+    }
+
+    getConfig (): Object {
+        return this.config;
+    }
+
+    get (path: string, defaultValue?: any): any {
+        return get(this.config, path, defaultValue);
+    }
+
+    getLocal () {
+        return JSON.parse(localStorage.getItem(this.appName) || '{}');
+    }
+
+    setLocal (path, value) {
+        const currentValue = this.getLocal();
+        set(currentValue, path, value);
+        localStorage.setItem(this.appName, JSON.stringify(currentValue));
+        this.config = merge(this.config, currentValue);
+    }
+
+    load (): Promise<Object> {
+        const useRemote = get(this._environment, 'remoteConfigs.enabled', false);
+
+        if (useRemote) {
+            this.logger.debug('load remote configuration');
+
+            const origin = get(this._environment, 'remoteConfigs.origin', '');
+            const urls = get(this._environment, 'remoteConfigs.paths', [])
+                .filter(Boolean)
+                .map(path => `${origin}${path}`);
+            return this.loadRemote(urls);
+        }
+
+        this.logger.debug('configuration ready');
+
+        return new Promise((resolve) => {
+            setTimeout(() => {
+                resolve(this);
+            }, 3000);
```

Comment:
This just a placeholder for now?

Reply:
Sort of, if there's no remote configuration there's no delay so the users won't see the loading screen. It might be too much, reducing to 500.